pkg/kedge/grpc/director/director.go

```diff
@@ -17,17 +18,17 @@ import (
 
 // New builds a StreamDirector based off a backend pool and a router.
 func New(pool backendpool.Pool, router router.Router) proxy.StreamDirector {
-	return func(ctx context.Context, fullMethodName string) (*grpc.ClientConn, error) {
+	return func(ctx context.Context, fullMethodName string) (context.Context, *grpc.ClientConn, error) {
 		beName, err := router.Route(ctx, fullMethodName)
 		if err != nil {
-			return nil, err
+			return ctx, nil, err
 		}
+
+		ctx = grpcutils.CloneIncomingToOutgoingMD(ctx)
+
 		grpc_ctxtags.Extract(ctx).Set("grpc.proxy.backend", beName)
```

Comment:
you want this to be before the clone ;)

Reply:
+1

Reply:
will fix

Reply:
https://github.com/improbable-eng/kedge/pull/87 lmao